ballista/core/src/execution_plans/shuffle_reader.rs

```diff
@@ -203,11 +204,11 @@ fn stats_for_partitions(
 }
 
 struct LocalShuffleStream {
-    reader: FileReader<File>,
+    reader: StreamReader<BufReader<File>>,
```

Comment:
https://docs.rs/arrow-ipc/latest/arrow_ipc/reader/struct.StreamReader.html#method.try_new

Seems this is buffered already.

Reply:
I don't disagree, but it is what our fork of arrow-rs is using right now: https://github.com/coralogix/arrow-rs/blob/dbdbb6b3427f85e987c494c68bef64af5435776a/arrow-ipc/src/reader.rs#L853